packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss

```diff
@@ -29,6 +27,11 @@ $unselectedHoverColor: $ms-color-neutralLighter;
   &.ms-DetailsRow.is-selected {
     background: $selectedColor;
   }
+
+  &.prevent-selection {
```

Comment:
If this class were on the List rather than DetailsRow then `is-multipleSelect` might work. On the details row I guess maybe `is-contentUnselectable`? Or disable selection by default and apply a class of `is-contentSelectable` to enable it.